connect_ext_test.go

```diff
@@ -2093,47 +2093,114 @@ func TestStreamUnexpectedEOF(t *testing.T) {
 			return
 		}
 		_, _ = io.Copy(io.Discard, request.Body)
-		header := responseWriter.Header()
-		header.Set("Content-Type", "application/connect+json")
 		testcase(responseWriter, request)
 	})
 	server := httptest.NewUnstartedServer(mux)
 	server.EnableHTTP2 = true
 	server.StartTLS()
 	t.Cleanup(server.Close)
 
-	client := pingv1connect.NewPingServiceClient(
-		server.Client(),
-		server.URL,
-		connect.WithProtoJSON(),
-	)
 	head := [5]byte{}
 	payload := []byte(`{"number": 42}`)
 	binary.BigEndian.PutUint32(head[1:], uint32(len(payload)))
 	testcases := []struct {
 		name       string
 		handler    http.HandlerFunc
+		options    []connect.ClientOption
 		expectCode connect.Code
 		expectMsg  string
 	}{{
-		name: "stream_unexpected_eof",
-		handler: func(responseWriter http.ResponseWriter, request *http.Request) {
+		name:    "connect_missing_end",
+		options: []connect.ClientOption{connect.WithProtoJSON()},
+		handler: func(responseWriter http.ResponseWriter, _ *http.Request) {
+			header := responseWriter.Header()
+			header.Set("Content-Type", "application/connect+json")
 			_, _ = responseWriter.Write(head[:])
 			_, _ = responseWriter.Write(payload)
 		},
 		expectCode: connect.CodeUnknown,
 		expectMsg:  "unknown: unexpected EOF",
 	}, {
-		name: "stream_partial_payload",
-		handler: func(responseWriter http.ResponseWriter, request *http.Request) {
+		name:    "grpc_missing_end",
+		options: []connect.ClientOption{connect.WithProtoJSON(), connect.WithGRPC()},
+		handler: func(responseWriter http.ResponseWriter, _ *http.Request) {
+			header := responseWriter.Header()
+			header.Set("Content-Type", "application/grpc+json")
+			_, _ = responseWriter.Write(head[:])
+			_, _ = responseWriter.Write(payload)
+		},
+		expectCode: connect.CodeInternal,
+		expectMsg:  "internal: gRPC protocol error: no Grpc-Status trailer",
+	}, {
+		name:    "grpc-web_missing_end",
+		options: []connect.ClientOption{connect.WithProtoJSON(), connect.WithGRPCWeb()},
+		handler: func(responseWriter http.ResponseWriter, _ *http.Request) {
+			header := responseWriter.Header()
+			header.Set("Content-Type", "application/grpc-web+json")
+			_, _ = responseWriter.Write(head[:])
+			_, _ = responseWriter.Write(payload)
+		},
+		expectCode: connect.CodeInternal,
+		expectMsg:  "internal: gRPC protocol error: no Grpc-Status trailer",
```

Comment:
Should this be `internal: gRPC-Web protocol error: no Grpc-Status trailer`?

Reply:
I believe the reason that it isn't is because internally gRPC and gRPC-web are nearly the same codepath, so the message just comes from the same line of code.

Reply:
Yeah, I'm not sure if it's worth changing but wanted to flag. Will resolve for now.

Reply:
Probably fine to change to "internal: protocol error: no Grpc-Status trailer"